plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt

```diff
@@ -79,7 +80,42 @@ abstract class LoginBrowser(
 
     protected val selectionSettings = mutableMapOf<String, BearerConnectionSelectionSettings>()
 
-    protected val onPendingToken: (InteractiveBearerTokenProvider) -> Unit = { provider ->
+    private var browserOpenTimer: Timer? = null
+
+    private fun startBrowserOpenTimer(credentialSourceId: CredentialSourceId) {
+        browserOpenTimer = Timer()
+        browserOpenTimer?.schedule(
+            object : TimerTask() {
+                override fun run() {
+                    AwsTelemetry.loginWithBrowser(
+                        project = null,
+                        credentialStartUrl = SONO_URL,
+                        result = Result.Failed,
+                        reason = "Browser authentication idle for more than 15min",
+                        credentialSourceId = credentialSourceId
+                    )
+                    AuthTelemetry.addConnection(
+                        result = Result.Failed,
+                        reason = "Browser authentication idle for more than 15min",
+                        credentialSourceId = credentialSourceId
+                    )
+                    stopBrowserOpenTimer()
+                }
+            },
```

Comment:
does SAM simplification not work?
```suggestion
            {
                AwsTelemetry.loginWithBrowser(
                    project = null,
                    credentialStartUrl = SONO_URL,
                    result = Result.Failed,
                    reason = "Browser authentication idle for more than 15min",
                    credentialSourceId = credentialSourceId
                )
                AuthTelemetry.addConnection(
                    result = Result.Failed,
                    reason = "Browser authentication idle for more than 15min",
                    credentialSourceId = credentialSourceId
                )
                stopBrowserOpenTimer()
            },
```

Reply:
Doesn't work
```
None of the following functions can be called with the arguments supplied.

schedule(TimerTask!, Date!) defined in java.util.Timer
schedule(TimerTask!, Long) defined in java.util.Timer
```

Reply:
oh because it's an abstract class ok